### DISCUSSION TITLE
Can't run serverless functions locally with `vercel dev`

### DISCUSSION THREAD

**Discussion by FrancescoBia** [vercel/community#637] [Help] [answered]
Hi! I'm trying to run `vercel dev` with my custom react + webpack app (therefore not using next or CRA default config). I'm also trying to add some serverless functions to the same project but I'm having some issues when running them locally.

I correctly added the `/api` folder at the root of the project. If I try to run `vercel dev` with **no custom dev script override**, the Vercel CLI picks up `npm run build` as a script, then build the project with webpack, and lastly starts it in localhost together with the functions. All good so far.

However, being a webpack app I rely on webpack-dev-server for quick rebuilds (and more), so I want to add a custom dev script to the Vercel project dashboard. Here is where the problems start:
- if I try to add `npm run dev` (my custom dev script that starts WDS) as dev script in the dashboard, the CLI correctly spins up my project via webpack-dev-server, however it never launches the functions.
- even more curious, if I try to manually pass `npm run build` to Vercel as a dev script, the CLI still doesn't boots up my functions. This is interesting as it means it's not an issue related to my specific dev script, but entirely (?) on how `vercel dev` works.

There's not much info in the documentation, however, there's an [article about the release](https://vercel.com/blog/vercel-dev) of `vercel dev` that states: 
> Once that [the build step] is completed, the tool ([...]) is invoked automatically.

In the examples, it seems that the CLI first builds the project and the functions, and then runs the dev script (which would be ideal for me).

That being said, what am I getting wrong? (happy to provide more context on the configuration if needed)

**Comment by swarnava** [comment, marked as answer]
Hi, Please reach out to support team with A [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) repository that can be shared publicly as a git repository so that our team can take a look.
